src/mqt/bench/utils.py

```diff
@@ -342,3 +343,73 @@ def postprocess_single_oqc_file(filename: str):
 
 def create_zip_file():
     return subprocess.call(f"zip -rj {get_zip_file_path()} {get_qasm_output_path()}", shell=True)
+
+
+def calc_qubit_index(qargs: list[Any], qregs: list[QuantumRegister], index: int) -> Any:
+    offset = 0
+    for reg in qregs:
+        if qargs[index] not in reg:
+            offset += reg.size
+        else:
+            qubit_index = offset + reg.index(qargs[index])
+            return qubit_index
+    error_msg = "Global qubit index for given local qubit index not found."
+    raise ValueError(error_msg)
+
+
+def calc_supermarq_features(
+    qc: QuantumCircuit,
+) -> tuple[float, float, float, float, float]:
```

Comment:
Also a nice opportunity to introduce a TypedDict, or probably even better: a dataclass.

Reply:
Implemented it.